### ISSUE TITLE
`WPS324` does not allow constant `None` function

### ISSUE THREAD

**Issue by phagenlocher** [wemake-services/wemake-python-styleguide#3531]
### What's wrong

`WPS324` does not allow a function that returns a constant `None`.

That's due to rules 2 and 3:

> 1. If any return has a value, all return nodes should have a value
> 2. Do not place return without a value at the end of a function
> 3. Do not use return None where just return is good enough

This is a problem when you have an abstract base class that with a method that returns `None` that you want to implement with a no-op. 

````python
class Abstract(ABC):
    def do_something(self: Self) -> None:
        raise NotImplementedError()

class Implementation(Abstract):
    def do_something(self: Self) -> None:
        return None # WPS324 fails (also without the None)
````

`pass` is disallowed by `WPS420`. The only solution I have found is using the `Ellipsis` object:

````python
class Implementation(Abstract):
    def do_something(self: Self) -> None:
        ...
````

### How it should be

I guess I am really just asking the question: Is using `Ellipsis` really the preferred way? In this case I believe it would be better to `return None` or `return` as it makes it explicit that this should be a no-op. 

My proposal would be that the rule should make an exception if the function is a constant `None` function without any other logic.

### Flake8 version and plugins

````
{
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.12.11",
    "system": "Linux"
  },
  "plugins": [
    {
      "plugin": "darglint",
      "version": "1.8.1"
    },
    {
      "plugin": "flake8-annotations",
      "version": "3.1.1"
    },
    {
      "plugin": "flake8-bugbear",
      "version": "24.12.12"
    },
    {
      "plugin": "flake8-builtins",
      "version": "2.5.0"
    },
    {
      "plugin": "flake8-class-attributes-order",
      "version": "0.3.0"
    },
    {
      "plugin": "flake8-comprehensions",
      "version": "3.16.0"
    },
    {
      "plugin": "flake8-docstrings",
      "version": "1.7.0"
    },
    {
      "plugin": "flake8-eradicate",
      "version": "1.5.0"
    },
    {
      "plugin": "flake8-executable",
      "version": "2.1.3"
    },
    {
      "plugin": "flake8-isort",
      "version": "6.1.2"
    },
    {
      "plugin": "flake8-length",
      "version": "0.3.1"
    },
    {
      "plugin": "flake8-max-function-length",
      "version": "0.9.0"
    },
    {
      "plugin": "flake8_cognitive_complexity",
      "version": "0.1.0"
    },
    {
      "plugin": "mccabe",
      "version": "0.7.0"
    },
    {
      "plugin": "pep8-naming",
      "version": "0.15.1"
    },
    {
      "plugin": "pycodestyle",
      "version": "2.14.0"
    },
    {
      "plugin": "pyflakes",
      "version": "3.4.0"
    },
    {
      "plugin": "wemake-python-styleguide",
      "version": "1.3.0"
    }
  ],
  "version": "7.3.0"
}
````

### pip information

````
pip 25.2 from /usr/local/lib/python3.12/site-packages/pip (python 3.12)
aiofiles==24.1.0
aiohappyeyeballs==2.6.1
aiohttp==3.12.15
aiosignal==1.4.0
alembic==1.16.5
annotated-types==0.7.0
anthropic==0.68.0
anyio==4.10.0
arrow==1.3.0
asyncer==0.0.8
atlassian-python-api==4.0.4
attrs==25.3.0
backoff==2.2.1
bcrypt==4.3.0
beautifulsoup4==4.13.5
bidict==0.23.1
black==25.1.0
blinker==1.9.0
build==1.3.0
cachetools==5.5.2
certifi==2025.8.3
certipy==0.2.2
cffi==2.0.0
chainlit==2.6.2
chardet==5.2.0
charset-normalizer==3.4.3
-e git+ssh://git@github.com/natuvion-ai/compose-ai.git@02e48fd609f15e39b3f156b5bb357debffd6be8b#egg=chatbot&subdirectory=apps/chatbot
chevron==0.14.0
Chunking==0.0.2
click==8.2.1
cognitive_complexity==1.3.0
colorama==0.4.6
coloredlogs==15.0.1
-e git+ssh://git@github.com/natuvion-ai/compose-ai.git@02e48fd609f15e39b3f156b5bb357debffd6be8b#egg=common&subdirectory=apps/common
cryptography==46.0.1
cuid==0.4
darglint==1.8.1
dataclasses-json==0.6.7
DataProperty==1.1.0
debugpy==1.8.14
deepeval==3.3.9
Deprecated==1.2.18
distro==1.9.0
docker==7.1.0
dockerspawner==14.0.0
docstring_parser==0.17.0
eradicate==2.3.0
escapism==1.0.1
execnet==2.1.1
fastapi==0.115.14
fastembed==0.7.3
filelock==3.19.1
filetype==1.2.0
flake8==7.3.0
flake8-annotations==3.1.1
flake8-bugbear==24.12.12
flake8-builtins==2.5.0
flake8-class-attributes-order==0.3.0
flake8-comprehensions==3.16.0
flake8-docstrings==1.7.0
flake8-eradicate==1.5.0
flake8-executable==2.1.3
flake8-isort==6.1.2
flake8-length==0.3.1
flake8-max-function-length==0.9.0
flake8_cognitive_complexity==0.1.0
Flask==3.1.2
flatbuffers==25.2.10
fqdn==1.5.1
frozenlist==1.7.0
fsspec==2025.9.0
google-auth==2.40.3
google-genai==1.38.0
googleapis-common-protos==1.70.0
greenlet==3.2.4
grimp==3.11
grpcio==1.75.0
h11==0.16.0
h2==4.3.0
hf-xet==1.1.10
hpack==4.1.0
httpcore==1.0.9
httpx==0.28.1
httpx-sse==0.4.1
huggingface-hub==0.35.0
humanfriendly==10.0
Hypercorn==0.17.3
hyperframe==6.1.0
hypothesis==6.135.29
idna==3.10
import-linter==2.3
importlib_metadata==8.7.0
inflection==0.5.1
iniconfig==2.1.0
isoduration==20.11.0
isort==6.0.1
itsdangerous==2.2.0
Jinja2==3.1.6
jiter==0.11.0
jmespath==1.0.1
joblib==1.5.2
jsonpatch==1.33
jsonpointer==3.0.0
jsonschema==4.25.1
jsonschema-specifications==2025.9.1
jupyter-events==0.12.0
jupyterhub==5.3.0
jupyterhub-nativeauthenticator==1.3.0
kaleido==0.2.1
langchain==0.3.27
langchain-community==0.3.27
langchain-core==0.3.76
langchain-experimental==0.3.4
langchain-ollama==0.3.6
langchain-qdrant==0.2.0
langchain-text-splitters==0.3.11
langsmith==0.4.29
lark==1.2.2
Lazify==0.4.0
literalai==0.1.201
loguru==0.7.3
lxml==5.4.0
Mako==1.3.10
markdown-it-py==4.0.0
markdownify==1.1.0
MarkupSafe==3.0.2
marshmallow==3.26.1
mbstrdecoder==1.1.4
mccabe==0.7.0
mcp==1.14.1
mdurl==0.1.2
mmh3==5.2.0
monotonic==1.6
mpmath==1.3.0
multidict==6.6.4
mypy_extensions==1.1.0
narwhals==2.5.0
nest-asyncio==1.6.0
networkx==3.5
-e git+ssh://git@github.com/natuvion-ai/compose-ai.git@02e48fd609f15e39b3f156b5bb357debffd6be8b#egg=neuromap&subdirectory=apps/neuromap
nodeenv==1.9.1
numpy==2.3.3
numpydantic==1.6.9
nvidia-cublas-cu12==12.8.3.14
nvidia-cuda-cupti-cu12==12.8.57
nvidia-cuda-nvrtc-cu12==12.8.61
nvidia-cuda-runtime-cu12==12.8.57
nvidia-cudnn-cu12==9.7.1.26
nvidia-cufft-cu12==11.3.3.41
nvidia-cufile-cu12==1.13.0.11
nvidia-curand-cu12==10.3.9.55
nvidia-cusolver-cu12==11.7.2.55
nvidia-cusparse-cu12==12.5.7.53
nvidia-cusparselt-cu12==0.6.3
nvidia-nccl-cu12==2.26.2
nvidia-nvjitlink-cu12==12.8.61
nvidia-nvtx-cu12==12.8.55
oauthlib==3.3.1
ollama==0.5.1
onetimepass==1.0.1
onnxruntime==1.22.1
onnxruntime-gpu==1.21.0
openai==1.108.0
opentelemetry-api==1.37.0
opentelemetry-exporter-otlp==1.37.0
opentelemetry-exporter-otlp-proto-common==1.37.0
opentelemetry-exporter-otlp-proto-grpc==1.37.0
opentelemetry-exporter-otlp-proto-http==1.37.0
opentelemetry-instrumentation==0.58b0
opentelemetry-instrumentation-alephalpha==0.47.2
opentelemetry-instrumentation-anthropic==0.47.2
opentelemetry-instrumentation-bedrock==0.47.2
opentelemetry-instrumentation-chromadb==0.47.2
opentelemetry-instrumentation-cohere==0.47.2
opentelemetry-instrumentation-crewai==0.47.2
opentelemetry-instrumentation-google-generativeai==0.47.2
opentelemetry-instrumentation-groq==0.47.2
opentelemetry-instrumentation-haystack==0.47.2
opentelemetry-instrumentation-lancedb==0.47.2
opentelemetry-instrumentation-langchain==0.47.2
opentelemetry-instrumentation-llamaindex==0.47.2
opentelemetry-instrumentation-logging==0.58b0
opentelemetry-instrumentation-marqo==0.47.2
opentelemetry-instrumentation-mcp==0.47.2
opentelemetry-instrumentation-milvus==0.47.2
opentelemetry-instrumentation-mistralai==0.47.2
opentelemetry-instrumentation-ollama==0.47.2
opentelemetry-instrumentation-openai==0.47.2
opentelemetry-instrumentation-openai-agents==0.47.2
opentelemetry-instrumentation-pinecone==0.47.2
opentelemetry-instrumentation-qdrant==0.47.2
opentelemetry-instrumentation-redis==0.58b0
opentelemetry-instrumentation-replicate==0.47.2
opentelemetry-instrumentation-requests==0.58b0
opentelemetry-instrumentation-sagemaker==0.47.2
opentelemetry-instrumentation-sqlalchemy==0.58b0
opentelemetry-instrumentation-threading==0.58b0
opentelemetry-instrumentation-together==0.47.2
opentelemetry-instrumentation-transformers==0.47.2
opentelemetry-instrumentation-urllib3==0.58b0
opentelemetry-instrumentation-vertexai==0.47.2
opentelemetry-instrumentation-watsonx==0.47.2
opentelemetry-instrumentation-weaviate==0.47.2
opentelemetry-instrumentation-writer==0.47.2
opentelemetry-proto==1.37.0
opentelemetry-sdk==1.37.0
opentelemetry-semantic-conventions==0.58b0
opentelemetry-semantic-conventions-ai==0.4.13
opentelemetry-util-http==0.58b0
orjson==3.11.3
packaging==25.0
pamela==1.2.0
pandas==2.2.3
pathlib==1.0.1
pathspec==0.12.1
pathvalidate==3.3.1
pep8-naming==0.15.1
pillow==11.3.0
pip-tools==7.4.1
platformdirs==4.4.0
plotly==6.0.1
pluggy==1.6.0
portalocker==3.2.0
posthog==3.25.0
priority==2.0.0
prometheus_client==0.23.1
propcache==0.3.2
protobuf==6.32.1
py_rust_stemmers==0.1.5
pyasn1==0.6.1
pyasn1_modules==0.4.2
pycodestyle==2.14.0
pycparser==2.23
pydantic==2.11.5
pydantic-settings==2.10.1
pydantic_core==2.33.2
pydocstyle==6.3.0
pyfiglet==1.0.4
pyflakes==3.4.0
Pygments==2.19.2
PyJWT==2.10.1
pyproject_hooks==1.2.0
pyright==1.1.403
pytablewriter==1.2.1
pytest==8.4.1
pytest-asyncio==1.1.0
pytest-github-actions-annotate-failures==0.3.0
pytest-md-report==0.7.0
pytest-repeat==0.9.4
pytest-rerunfailures==12.0
pytest-xdist==3.8.0
python-dateutil==2.9.0.post0
python-dotenv==1.1.1
python-engineio==4.12.2
python-json-logger==3.3.0
python-multipart==0.0.18
python-socketio==5.13.0
pytz==2025.2
PyYAML==6.0.2
qdrant-client==1.15.1
Quart==0.20.0
redis==6.2.0
referencing==0.36.2
regex==2025.9.18
requests==2.32.3
requests-oauthlib==2.0.0
requests-toolbelt==1.0.0
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rfc3987-syntax==1.1.0
rich==14.1.0
rpds-py==0.27.1
rsa==4.9.1
safetensors==0.6.2
scikit-learn==1.7.2
scipy==1.16.2
sentence-transformers==4.1.0
sentry-sdk==2.38.0
setuptools==75.8.0
shellingham==1.5.4
simple-websocket==1.1.0
six==1.17.0
sniffio==1.3.1
snowballstemmer==3.0.1
sortedcontainers==2.4.0
soupsieve==2.8
SQLAlchemy==2.0.43
sse-starlette==3.0.2
starlette==0.46.2
sympy==1.14.0
syncer==2.0.3
tabledata==1.3.4
tabulate==0.9.0
tcolorpy==0.1.7
temporalio==1.11.1
tenacity==9.1.2
threadpoolctl==3.6.0
tokenizers==0.22.0
tomli==2.2.1
torch==2.7.1+cu128
tornado==6.5.2
tqdm==4.67.1
traceloop-sdk==0.47.2
traitlets==5.14.3
transformers==4.56.1
triton==3.3.1
typepy==1.3.4
typer==0.17.4
types-protobuf==6.32.1.20250918
types-python-dateutil==2.9.0.20250822
typing-inspect==0.9.0
typing-inspection==0.4.1
typing_extensions==4.15.0
tzdata==2025.2
uri-template==1.3.0
urllib3==2.5.0
uvicorn==0.35.0
watchfiles==0.20.0
webcolors==24.11.1
websockets==15.0.1
wemake-python-styleguide==1.3.0
Werkzeug==3.1.3
wheel==0.45.1
wrapt==1.17.3
wsproto==1.2.0
yarl==1.20.1
zipp==3.23.0
zstandard==0.25.0
````

### OS information

`Linux 8f5da943fd9a 6.8.0-1029-aws #31~22.04.1-Ubuntu SMP Thu Apr 24 21:16:18 UTC 2025 x86_64 GNU/Linux`

**Comment by sobolevn** [comment]
```python
class Abstract(ABC):
    def do_something(self: Self) -> None:
        raise NotImplementedError()

class Implementation(Abstract):
    def do_something(self: Self) -> None:
        """It is explicty empty and does nothing."""
```

👍
